test/functional/blockprod_generate_pos_blocks.py

```diff
@@ -0,0 +1,372 @@
+                            new_pool_id,
+                            {
+                                "value": "10000000000000000",
+                                "staker": {
+                                    "PublicKey": new_stake_public_key,
+                                },
+                                "vrf_public_key": new_vrf_public_key,
+                                "decommission_key": "AnyoneCanSpend",
+                                "margin_ratio_per_thousand": 1000,
+                                "cost_per_block" : "0"
+                            },
+                        ],
+                    },
+                ],
+            },
+            "signatures": [
+                {
+                    "NoSignature": None,
+                },
+            ],
+        }
+
+        (create_new_pool_transaction_encoded, create_new_pool_transaction_id) = self.pack_transaction(create_new_pool_transaction)
+        self.generate_block(block_input_data, [create_new_pool_transaction_encoded])
+
+        #
+        # Stake many blocks with the new stake pool
+        #
+
+        kernel_input_outpoint = {
+            "id": {
+                "Transaction": self.hex_to_dec_array(create_new_pool_transaction_id),
+            },
+            "index": 0,
+        }
+
+        kernel_input_utxo = {
+            "CreateStakePool": [
+                new_pool_id,
+                {
+                    "value": "10000000000000000",
+                    "staker": {
+                        "PublicKey": new_stake_public_key,
+                    },
+                    "vrf_public_key": new_vrf_public_key,
+                    "decommission_key": "AnyoneCanSpend",
+                    "margin_ratio_per_thousand": 1000,
+                    "cost_per_block" : "0"
+                },
+            ],
+        }
+
+        for _ in range(1, random.randint(1,100)):
+            block_input_data = block_input_data_obj.encode({
+                "PoS": {
+                "stake_private_key": new_stake_private_key,
+                    "vrf_private_key": new_vrf_private_key,
+                    "pool_id": new_pool_id,
+                    "kernel_inputs": [
+                        {
+                            "Utxo": kernel_input_outpoint,
+                        },
+                    ],
+                    "kernel_input_utxo": [kernel_input_utxo]
+                }
+            }).to_hex()[2:]
+
+            self.generate_block(block_input_data, [])
+
+            kernel_input_outpoint = {
+                "id": {
+                    "BlockReward": self.hex_to_dec_array(self.nodes[0].chainstate_best_block_id()),
+                },
+                "index": 0,
+            }
+
+            kernel_input_utxo = {
+                "ProduceBlockFromStake": [
+                    {
+                        "PublicKey": new_stake_public_key,
+                    },
+                    new_pool_id,
+                ],
+            }
+
```

Comment:
Could you also pull blocks by height (all the ones you expect to have been created), and verify that they're PoS in ConsensusData? Then verify it's constituting a chain (every previous block is the one before).

First you get the block id by height, then you pull the block by the id you got.